jac/jaclang/compiler/parser.py

```diff
@@ -852,41 +845,29 @@ def ability_decl(self, kid: list[ast.AstNode]) -> ast.Ability:
             ability_decl: KW_OVERRIDE? KW_STATIC? KW_CAN access_tag? STRING?
                 named_ref (func_decl | event_clause) (code_block | SEMI)
             """
-            chomp = [*kid]
-            is_override = (
-                isinstance(chomp[0], ast.Token) and chomp[0].name == Tok.KW_OVERRIDE
-            )
-            chomp = chomp[1:] if is_override else chomp
-            is_static = (
-                isinstance(chomp[0], ast.Token) and chomp[0].name == Tok.KW_STATIC
+            signature: ast.FuncSignature | ast.EventSignature | None = None
+            is_override = self.match_token(Tok.KW_OVERRIDE) is not None
+            is_static = self.match_token(Tok.KW_STATIC) is not None
+            self.consume_token(Tok.KW_CAN)
+            access = self.match(ast.SubTag)
+            semstr = self.match(ast.String)
+            name = self.consume(ast.NameAtom)
+            signature = self.match(ast.FuncSignature) or self.consume(
+                ast.EventSignature
+            )
+            body = self.match(ast.SubNodeList)
```

Comment:
Make sure to consume SEMI if the body is none. Read bellow for the reasoning for processing all valid rules/ tokens.